localstack/services/secretsmanager/provider.py

```diff
@@ -574,61 +582,75 @@ def backend_rotate_secret(
     # version as AWSCURRENT then any later invocation of RotateSecret assumes
     # that a previous rotation request is still in progress and returns an error.
     try:
+        pending_version = None
         version = next(
             version
             for version in secret.versions.values()
             if AWSPENDING in version["version_stages"]
         )
-        if AWSCURRENT in version["version_stages"]:
+        if AWSCURRENT not in version["version_stages"]:
             msg = "Previous rotation request is still in progress."
-            raise InvalidRequestException(msg)
+            # Delay exception, so we can trigger lambda again
+            pending_version = [InvalidRequestException(msg), version]
 
     except StopIteration:
         # Pending is not present in any version
         pass
 
-    # Begin the rotation process for the given secret by invoking the lambda function.
-    #
-    # We add the new secret version as "pending". The previous version remains
-    # as "current" for now. Once we've passed the new secret through the lambda
-    # rotation function (if provided) we can then update the status to "current".
-    new_version_id = self._from_client_request_token(client_request_token)
-    #
-    self._add_secret(
-        secret_id,
-        None,
-        description=secret.description,
-        tags=secret.tags,
-        version_id=new_version_id,
-        version_stages=[AWSPENDING],
-    )
     secret.rotation_lambda_arn = rotation_lambda_arn
-    if rotation_rules:
-        secret.auto_rotate_after_days = rotation_rules.get(rotation_days, 0)
+    secret.auto_rotate_after_days = rotation_rules.get(rotation_days, 0)
     if secret.auto_rotate_after_days > 0:
+        secret.next_rotation_date = int(time.time()) + (int(rotation_period) * 86400)
```

Comment:
```suggestion
        wait_interval_s = int(rotation_period) * 86400
        secret.next_rotation_date = int(time.time()) + wait_interval_s
```

nit: splitting this makes the behavior clearer to the reader